olp-cpp-sdk-dataservice-read/src/ProtectDependencyResolver.cpp

```diff
@@ -61,7 +61,8 @@ ProtectDependencyResolver::GetKeysToProtect(const TileKeys& tiles) {
 
 ProtectDependencyResolver::QuadsType::iterator
 ProtectDependencyResolver::FindQuad(const geo::TileKey& tile_key) {
-  auto max_depth = std::min<std::int32_t>(tile_key.Level(), kQuadTreeDepth);
+  auto max_depth = std::min<std::int32_t>(
```

Comment:
tile_key.Level() is unsigned, kQuadTreeDepth is whatever you want, so why does max_depth need to be signed when the below loop can also be inverted and instead of decrementing you can increment.